src/main/java/org/opensearch/sdk/handlers/ExtensionsInitRequestHandler.java

```diff
@@ -65,7 +66,10 @@ public InitializeExtensionResponse handleExtensionInitRequest(InitializeExtensio
             extensionsRunner.setExtensionNode(extensionInitRequest.getExtension());
             // TODO: replace with sdkTransportService.getTransportService()
             TransportService extensionTransportService = extensionsRunner.getExtensionTransportService();
-            extensionTransportService.connectToNode(extensionsRunner.opensearchNode);
+            extensionTransportService.connectToNodeAsExtension(
```

Comment:
Without this change, it causes a fatal error to the opensearch node and the node fails with the error:

```
Likely root cause: OpenSearchException[Transport client authentication no longer supported.]
```

The fix for this is the companion PR in core that introduces `connectToNodeAsExtension`. Inside that function it injects `extension_unique_id` into the thread context of the connection validator. I'm still wrapping my head around this change, but it solves for the fatal error.

Reply:
@saratvemulapalli I'm looking into this more today if this is what's blocking this PR. This was the only way I could figure out how to get the handshake working between a node and an extension with the security plugin installed. I will look more in detail to see how cross cluster works to see if this can more closely model cross cluster cases.

Reply:
Thanks @cwperks. It makes sense.

Reply:
@cwperks just trying to understand why `connectToNodeAsExtension` is required? Is it because we are injecting the unique_id in the thread context and since no authentication was before so we didn't face any problem with `connectToNode`?

Reply:
@owaiskazi19 You haven't faced this problem yet because you haven't been testing with the security plugin installed. The security plugin has logic that forbids transport requests from anything that isn't a node in the cluster and an extension is not a node in the cluster.

What I was doing with `connectToNodeAsExtension` and also with line 51 in this file is to identify all of these transport requests as coming from an extension and the extension is identified by its unique id. For some reason line 51 was not working with the handshake so I created `connectToNodeAsExtension` which also injects the unique id into the thread context of the connection validator. See the linked PR to the security plugin in the PR description, it checks if the experimental feature flag for extensions is enabled and then checks to see if a transport request comes from an extension so it can permit the request.